Lesson 5 1

Hermann27 opened the pull request:
Please, review my assignment whenever you have time. Thank you for your guidance.

Hermann27 committed:
clean up some file

Hermann27 committed:
Installed Prop Types Library

Hermann27 committed:
On This commit, I have installed Proptypes using this cmd : npm install prop-types.

The had to implement the Protypes on all my components

Hermann27 committed:
Change my Proptype emplacement.

Hermann27 committed:
Update CSS style on index.css amd TodoListItem.module.css file

Hermann27 committed:
Cleanup my index.html to remove anything that was added.

Hermann27 committed:
On this commit, I have implemented:
1. Sort my data from my airtable using JavaScript inside the fectData func.

2. Created a toggle button so the user can switch between asc and desc sort order.

3.Automatically sorted my data when added.

AnnaSolovykh commented on src/App.jsx:
```diff
@@ -113,6 +158,10 @@ function App() {
               <center>
```
Instead of using the` <center> `tag, which is deprecated, you can achieve the same result using CSS.

Hermann27 replied:
Thank you for the feedback. Completed!

AnnaSolovykh commented on src/App.jsx:
```diff
@@ -100,6 +137,14 @@ function App() {
+
+  React.useEffect(() => {
+    fetchData();
+  }, [isAscending]); // Refetch data when sort order changes
```
The fetchData function is being called in two useEffect hooks, which might not be necessary. You can refactor this to avoid re-fetching the data unnecessarily when toggling the sort order.

You can manage the sorting on the client-side without refetching the data when toggling the sort order. Instead of fetching data again when isAscending changes, you can sort the existing todoList state directly.

Hermann27 replied:
Completed

AnnaSolovykh commented:
As for rest, looks good to me!

Hermann27 committed:
Following changed have been added on this commit:
1. use css module on the app.css to center my cotainer.

2.removed the redomdance of useEffect on App.jxs

3. Refactored the code (re-structure and removed unnecessary code that was commented).

AnnaSolovykh commented:
Hi Hermann! Good job, but looks like smth is wrong with sorting. 
<img width="932" alt="Screenshot 2024-10-07 at 10 16 34" src="https://github.com/user-attachments/assets/864689cf-2466-4025-8c63-1d27238d2628">
<img width="948" alt="Screenshot 2024-10-07 at 10 16 47" src="https://github.com/user-attachments/assets/5c8c6668-951f-4e74-a0f3-2f2fa124d74a">

I am sorting by created time as the lesson is asking for. Do you want me to do by Title

Hermann27 committed:
This updates fixed the mixing createdTime field on the display Table.

Hermann27 committed:
This commit resolved:
1. Formatting the Create Time field for a better format.

2. Update some style button.
3. Added an update button only.
